lib/Core/Executor.cpp

```diff
@@ -498,7 +518,7 @@ MemoryObject * Executor::addExternalObject(ExecutionState &state,
 extern void *__dso_handle __attribute__ ((__weak__));
 
 void Executor::initializeGlobals(ExecutionState &state) {
-  Module *m = kmodule->module;
+  Module *m = kmodule->module.get();
```

Comment:
What's the point of unqiue pointer of `.get` is used?

Reply:
The point is ownership. `unique_ptr`/`shared_ptr` make clear who owns what. Still, sometimes you need to work on an object and for that you typically use a reference. But the contract is now that you won't delete that object nor do you have to do that nor do you try to take ownership by keeping a reference longer alive than the owner itself.

(I admit KLEE's codebase is not there yet but we have to start somewhere ;) )

Reply:
I don't understand why you can't use a reference to either `Module` or `unique_ptr<Module>` here?